### PR TITLE
feat: Added Counter disputes

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [X] I ran `npm run re:build`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+      let copyFileUploadedDict = fileUploadedDict->Dict.copy
+
+      response->Array.forEachWithIndex((ele, index) => {
+        let keyValue = keyFromDictArray[index]->Option.getOr("")
+        let dictValue = fileUploadedDict->getDictfromDict(keyValue)
+        switch Js.Json.classify(ele) {
+        | Js.Json.JSONObject(jsonDict) => {
+            let fileId = jsonDict->getString("file_id", "")
+            dictValue->Dict.set("fileId", fileId->Js.Json.string)
+            copyFileUploadedDict->Dict.set(keyValue, dictValue->Js.Json.object_)
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:131]: Core API

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+    let onEvidenceSubmit = async () => {
+      try {
+        let url = getURL(~entityName=DISPUTES_ATTACH_EVIDENCE, ~methodType=Post, ())
+        let body = constructDisputesBody(fileUploadedDict, disputeId)
+        let response = await updateDetails(url, body->JSON.Encode.object, Post, ())
+        setDisputeData(_ => response)
+        setDisputeEvidenceStatus(_ => EvidencePresent)
+      } catch {
+      | Exn.Error(e) => {
+          let _err = Exn.message(e)->Option.getOr("Failed to Fetch!")
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:232]: Use _

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+                    text="Submit your evidence"
+                    buttonSize={Small}
+                    onClick={_ => onEvidenceSubmit()->ignore}
+                  />
+                  <Button
+                    buttonType={Secondary}
+                    text="Attach More"
+                    buttonSize={Small}
+                    customButtonStyle="!bg-white"
+                    leftIcon={Euler("paper-clip")}
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:347]: Why Euler is still there?

File: src/utils/LogicUtils.res [hunk 4]
```diff
@@ -628,3 +628,26 @@ let listOfMatchedText = (text, searchText) => {
...
+    fileName
+  } else {
+    let truncatedText =
+      beforeDotFileName->String.slice(~start=0, ~end=maxTextLength)->String.concat("...")
+    truncatedText ++ "." ++ afterDotFileType
+  }
+}
+
+let getDaysDiffForDates = (~startDate, ~endDate) => {
+  let startDate = startDate->Js.Date.fromFloat
```
Comment (PritishBudhiraja) [inline src/utils/LogicUtils.res:647]: Core API

File: src/utils/LogicUtils.res [hunk 5]
```diff
@@ -628,3 +628,26 @@ let listOfMatchedText = (text, searchText) => {
...
+  let endDate = endDate->Js.Date.fromFloat
```
Comment (PritishBudhiraja) [inline src/utils/LogicUtils.res:648]: Core API

File: src/utils/LogicUtils.res [hunk 6]
```diff
@@ -628,3 +628,26 @@ let listOfMatchedText = (text, searchText) => {
...
+  let daysDiff = Math.abs(endDate->Js.Date.getTime -. startDate->Js.Date.getTime)
```
Comment (PritishBudhiraja) [inline src/utils/LogicUtils.res:649]: Core API

**Review by PritishBudhiraja — CHANGES_REQUESTED** [review]

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+
+      let dictToIterate = keyFromDictArray->Array.filter(ele => {
+        let keyObject = fileUploadedDict->getDictfromDict(ele)
+        keyObject->Dict.get("fileId")->Option.isNone
+      })
+
+      let promisesOfAttachEvidence = dictToIterate->Array.map(ele => {
+        let jsonObject = fileUploadedDict->Js.Dict.get(ele)->Option.getOr(Js.Json.null)
+        let fileValue = jsonObject->getDictFromJsonObject->getJsonObjectFromDict("uploadedFile")
+        let res = acceptFile(ele, fileValue)
```
[the remark this answers is not in the document]
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:117]: Assigning it to a variable and then returning it provides better code readability.

[the remark this answers is not in the document]
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:113]: When we are using that then only else it's just unnecessary overhead

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res]:
```
 | _ => {
   showToast(
```

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res (hunk 7 above)
[the remark this answers is not in the document]
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:114]: As this is return value from the function , its better to assign it to a variable and return

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res]: Update this also

```
| _ => {
 showToast(
            ~message=`Failed to submit the evidence. Try again !`,
            ~toastType=ToastError,
            (),
          )
}
```

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res [hunk 8]
```diff
@@ -0,0 +1,434 @@
...
+    ~setUploadEvidenceModal,
+    ~disputeId,
+    ~setDisputeEvidenceStatus,
+    ~fileUploadedDict,
+    ~setFileUploadedDict,
+  ) => {
+    open APIUtils
+    open LogicUtils
+    let updateDetails = useUpdateMethod()
+    let acceptFile = async (keyValue, fileValue) => {
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:84]: remove the async

File: src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res [hunk 9]
```diff
@@ -0,0 +1,434 @@
...
+      let url = getURL(~entityName=DISPUTES_ATTACH_EVIDENCE, ~methodType=Put, ())
+      let formData = formData()
+      append(formData, "dispute_id", disputeId)
+      append(formData, "evidence_type", keyValue)
+      append(formData, "file", fileValue)
+
+      let response = await updateDetails(
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/Disputes/UploadEvidenceForDisputes.res:91]: remove the await statement

**Review by PritishBudhiraja — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**